### PR TITLE
[rustdoc] Specific crate search

### LINKED ISSUE #54616: [rustdoc] Add possibility to search in a given crate only
We could add a dropdown menu to the search with the available crates and allow to look only in a given crate.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #54616

screen shot 2018-10-01 at 00 55 45

r? @QuietMisdreavus

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
@QuietMisdreavus: You can see it in action here: http://gtk-rs.org/docs/gtk/struct.Window.html

File: src/librustdoc/html/static/main.js [hunk 1]
```diff
@@ -1376,8 +1385,13 @@
             var ret_in_args = addTab(results['in_args'], query, false);
             var ret_returned = addTab(results['returned'], query, false);
 
+            var filter = "";
+            if (filterCrates !== undefined) {
+                filter = " (in <b>" + filterCrates + "</b> crate)";
```
Comment (QuietMisdreavus) [inline src/librustdoc/html/static/main.js]: Small wording nit:

```suggestion
                filter = " (in crate <b>" + filterCrates + "</b>)";
```

**Review by QuietMisdreavus — COMMENTED** [review]
Wow, this is nice! I think it'd be ideal if it was using the sans-serif font. (And as a bonus, could it also have a little magnifying glass icon? Something like https://www.fileformat.info/info/unicode/char/1f50d/index.htm, i guess. It seems like a little touch that would make it nicer.)

**Comment by GuillaumeGomez** [comment]
It's complicated because it takes already a lot of space and I'm worried that it might be an issue on mobile devices. However I intend to put arrows on the right of the (heavily modified) select box. That might help a bit.

**Comment by QuietMisdreavus** [comment]
Re-review note: the wording nit still stands, and there's now a merge conflict. It's okay if we can't include the little icon, but can we still use the sans-serif font? I feel like it would be better to use it there, since it's in more of a "heading" position than a "body text" position, where we use the serif font currently.

image

**Comment by GuillaumeGomez** [comment]
Sure, I didn't come back to this PR yet. :)

**Comment by TimNN** [comment]
Ping from triage @GuillaumeGomez!

**Comment by GuillaumeGomez** [comment]
Still the same.

**Comment by TimNN** [comment]
Ping from triage! This PR hasn't made any progress in a while, so I'm closing it per our [triage guidelines](https://forge.rust-lang.org/triage-procedure.html). Thanks for your contributions and please feel free to re-open in the future.

**Comment by GuillaumeGomez** [comment]
I added the arrow as you can see here:

screenshot 2018-11-29 at 01 29 11

**Pull request closed without merging**